### ISSUE TITLE
UVW 2026: Underthesea Vietnamese Wikipedia Dataset

### ISSUE THREAD

**Issue by rain1024** [undertheseanlp/underthesea#896]
# UVW 2026: Underthesea Vietnamese Wikipedia Dataset

A high-quality, cleaned dataset of Vietnamese Wikipedia articles enriched with Wikidata metadata. Designed for Vietnamese NLP research including language modeling, text generation, text classification, and model pretraining.

## Dataset

🤗 **HuggingFace:** [undertheseanlp/UVW-2026](https://huggingface.co/datasets/undertheseanlp/UVW-2026)

📦 **Repository:** [github.com/undertheseanlp/UVW-2026](https://github.com/undertheseanlp/UVW-2026)

## Statistics

| Property | Value |
|----------|-------|
| **Total Articles** | 1,118,224 |
| **Train / Validation / Test** | 894,579 / 111,822 / 111,823 |
| **Wikidata Coverage** | 99.4% |
| **Unique Categories** | 11,549 |
| **License** | CC BY-SA 4.0 |

## Features

- **Clean text**: Wikipedia markup, templates, references removed
- **Wikidata integration**: Articles linked to Wikidata entities with semantic categories
- **Quality scoring**: Each article scored 1-10 based on content quality metrics
- **Unicode normalized**: NFC normalization for consistent text processing
- **Ready to use**: Pre-split into train/validation/test sets (80/10/10)

## Schema

```json
{
  "id": "Việt_Nam",
  "title": "Việt Nam",
  "content": "Việt Nam, tên chính thức là Cộng hòa Xã hội chủ nghĩa Việt Nam...",
  "num_chars": 45000,
  "num_sentences": 500,
  "quality_score": 9,
  "wikidata_id": "Q881",
  "main_category": "quốc gia có chủ quyền"
}
```

## Quick Start

```python
from datasets import load_dataset

dataset = load_dataset("undertheseanlp/UVW-2026")
train = dataset["train"]
print(train[0])
```

## Top Categories

| Category | Count | % |
|----------|------:|--:|
| đơn vị phân loại (taxon) | 618,281 | 55.3% |
| người (human) | 78,191 | 7.0% |
| xã của Pháp | 35,635 | 3.2% |
| khu định cư | 20,276 | 1.8% |
| tiểu hành tinh | 17,891 | 1.6% |
| xã của Việt Nam | 7,088 | 0.6% |

## Pipeline

1. **Download** - Vietnamese Wikipedia XML dump from Wikimedia
2. **Extract** - Parse XML, clean markup (templates, refs, links, tables)
3. **Normalize** - Apply Unicode NFC normalization
4. **Score** - Calculate quality metrics (1-10)
5. **Enrich** - Add Wikidata IDs and categories via Wikidata API
6. **Split** - Create train/validation/test (80/10/10, seed=42)

## Related

- Part of the [Underthesea](https://github.com/undertheseanlp/underthesea) Vietnamese NLP project
